client/src/components/Modal/MapModal.js

```diff
@@ -0,0 +1,111 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { View, Modal, TouchableOpacity, Dimensions, TouchableWithoutFeedback, Text } from 'react-native';
+
+import MapView, { Marker } from 'react-native-maps';
+import { IconMarker } from '../../components/MapMarker/';
+import MapDelta from '../../selectors/MapDelta';
+
+import styles from './styles';
+
+const screen = Dimensions.get('window');
+
+const ASPECT_RATIO = screen.width / screen.height;
+const LATITUDE_DELTA = 0.0922;
+const LONGITUDE_DELTA = LATITUDE_DELTA * ASPECT_RATIO;
+
+class MapModal extends Component {
+  zoomMap = () => {
+    if (this.props.initialRegion) {
+      this.map.animateToRegion(
+        MapDelta(
+          this.props.initialRegion.locationLatitude,
+          this.props.initialRegion.locationLongitude,
+          2000, // 2000m height
+        ),
+      );
+    }
+  }
+
+  render() {
+    return (
+      <View>
+        <Modal
+          transparent
+          visible={this.props.visible}
+          onRequestClose={this.props.closeModal}
+          animationType="fade"
+        >
+          <TouchableWithoutFeedback onPress={this.props.closeModal}>
+            <View style={styles.overlayStyle}>
+              <View style={styles.headerContainer}>
+                <TouchableOpacity>
+                  <View style={styles.headerStyle}>
+                    <Text style={styles.headerTextStyle}>{this.props.title}</Text>
+                  </View>
+                </TouchableOpacity>
+              </View>
+              <View style={styles.mapContainer}>
+                <View>
+                  <View>
+                    {/* TouchableWithoutFeedback prevents touch propagation */}
+                    <TouchableWithoutFeedback>
+                      <MapView
+                        onMapReady={this.zoomMap}
+                        ref={(ref) => {
+                        this.map = ref;
```

Comment:
here until L77 the indentation is fucked up please fix

Reply:
got it